core/metrics/src/main/scala/org/typelevel/otel4s/metrics/ObservableInstrumentBuilder.scala

```diff
@@ -61,4 +63,19 @@ trait ObservableInstrumentBuilder[F[_], A, Instrument] {
   def createWithCallback(
       cb: ObservableMeasurement[F, A] => F[Unit]
   ): Resource[F, Instrument]
+
+  /** Creates an asynchronous instrument that an effect that produces a number
+    * of measurements
+    *
```

Comment:
```suggestion
  /** Creates an asynchronous instrument that uses an effect that produces a number
    * of measurements.
    * 
    * The measurement effect will be evaluted when the instrument is being observed.
    * 
    * The measurement effect is expected to abide by the following restrictions:
    *   - Short-living and (ideally) non-blocking
    *   - Run in a finite amount of time
    *   - Safe to call repeatedly, across multiple threads
```

A bit of a copy-paste from the `createWithCallback` Scaladoc.